File: SistemaTicket/sistema_ticketero/backend/src/Ticketero.Domain/Entities/Area.cs
namespace Ticketero.Domain.Entities;

public class Area
{
    public int Id { get; set; }
    public string Nombre { get; set; } = string.Empty;
    public string Prefijo { get; set; } = string.Empty;
    public bool Activo { get; set; } = true;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<User> Users { get; set; } = new List<User>();
    public ICollection<Ticket> Tickets { get; set; } = new List<Ticket>();
}

File: SistemaTicket/sistema_ticketero/lib/domain/entities/area.dart
class Area {
  final int id;
  final String nombre;
  final String prefijo;
  final String logoUrl;
  final bool activo;

  const Area({
    required this.id,
    required this.nombre,
    required this.prefijo,
    this.logoUrl = '',
    required this.activo,
  });

  Area copyWith({
    int? id,
    String? nombre,
    String? prefijo,
    String? logoUrl,
    bool? activo,
  }) {
    return Area(
      id: id ?? this.id,
      nombre: nombre ?? this.nombre,
      prefijo: prefijo ?? this.prefijo,
      logoUrl: logoUrl ?? this.logoUrl,
      activo: activo ?? this.activo,
    );
  }
}
